pylearn2/datasets/tests/test_retina.py

```diff
@@ -0,0 +1,79 @@
+"""
+Tests for ../retina.py
+"""
+import numpy as np
+
+from pylearn2.utils import as_floatX
+from pylearn2.datasets import dense_design_matrix
+from pylearn2.datasets.retina import (foveate_channel,
+                                      defoveate_channel,
+                                      get_encoded_size,
+                                      encode,
+                                      decode,
+                                      RetinaEncodingBlock,
+                                      RetinaDecodingBlock,
+                                      RetinaCodingViewConverter
+                                      )
+
+
+class test_retina():
+    """
+    Parameters
+    ----------
+    None
+
+    Notes
+    -----
+    Testing class that checks retina preprocessing functions
+    """
+    def test_foveate_channel(self):
+        """
+        Foveate_channel implicitly tests downsample_ring and downsample_rect
+        """
+        img = np.array([[[0.51507119, 0.71686988], [0.62864236, 0.75365718]]])
+        output = np.zeros((1, 4))
+        foveate_channel(img, [2, 3], output, 0)
+        np.testing.assert_allclose(output, np.array([[0.65356015, 0.65356015,
+                                                      0., 0.]]))
+
+    def test_get_encoded_size(self):
+        """
+        Test encoding size method
+        """
+        assert get_encoded_size(4, 12, [2]) == 12
+
+    def test_encode(self):
+        """
+        Test encode method
+        """
+        test_input = np.array([[[[0, 0], [1, 1]]]])
+        np.testing.assert_array_equal(encode(test_input, [1]),
+                                      np.array([[0., 1., 0., 1.]]))
+
+    def test_decode(self):
+        """
+        .. note::
+            It is ambiguous what this function is suppose to do based
+            on the documentation, so it is missing a test for now.
+            Testing it would also test defoveate_channel/restore_ring/
+            restore rec.
```

Comment:
So, it works for 96x96 images, but not for other sizes.
The reason is that there is still a "96" hardcoded in retina.py on line 236, it should be "img_w" instead.

With that fix, you can write a test where you sample a random encoded representation, then decode and encode it, and check that you recover (almost) the original random matrix.